src/js/messaging/components/MessageSearchAdvanced.jsx

```diff
@@ -7,15 +7,28 @@ import ErrorableTextInput from '../../common/components/form-elements/ErrorableT
 import ErrorableCheckbox from '../../common/components/form-elements/ErrorableCheckbox';
 
 class MessageSearchAdvanced extends React.Component {
+  constructor(props) {
+    super(props);
+    this.handleAdvancedSearchToggle = this.handleAdvancedSearchToggle.bind(this);
+  }
+
+  handleAdvancedSearchToggle() {
+    this.props.onAdvancedSearch();
+  }
+
   render() {
+    let advancedSearchForm;
+
     const toggleClass = classNames({
       fa: true,
-      'fa-chevron-up': this.props.isOpen,
-      'fa-chevron-down': !this.props.isOpen
+      'fa-chevron-up': this.props.isVisible,
+      'fa-chevron-down': !this.props.isVisible
     });
 
-    return (
-      <div className="msg-search-advanced">
+    if (this.props.isVisible === false) {
```

Comment:
This can just be one case (`if (this.props.isVisible)`). If false, it will already render nothing.